tests/phpunit/src/Gateway/PayplugPproTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use Payplug\PayplugWoocommerce\Controller\PayplugGenericGateway;
+use WC_Order;
+
+class PayplugPproTest extends TestCase
+{
+    public function testRefundNotAvailable()
```

Comment:
add integration tests (maybe in a different directory "integration" )
we should mock the payment_gateways and check if they obey the business rules they've.
When:

if the rules changed, we would have a error and we should fix the unit-testing to.
if the rules don't change and we by mistake update them, we will have an error saying also that we shouldn't had update them.